python/cudf/cudf/core/index.py

```diff
@@ -3542,13 +3553,37 @@ def __init__(
             self, ColumnAccessor({name: data}, verify=False)
         )
         self._freq = _validate_freq(freq)
-        if self._freq is not None:
+        # existing pandas index needs no additional validation
+        if self._freq is not None and not was_pd_index:
             unique_vals = self.to_series().diff().unique()
-            if len(unique_vals) > 2 or (
-                len(unique_vals) == 2
-                and unique_vals[1] != self._freq._maybe_as_fast_pandas_offset()
-            ):
-                raise ValueError("No unique frequency found")
+            if self._freq == cudf.DateOffset(months=1):
+                possible = pd.Series(
+                    [
+                        pd.NaT,
+                        pd.to_timedelta("28 days"),
```

Comment:
Do we have to account for leap years where feb has 29 days? I noticed a place below where we include 29 days in month-specific code as well.